packages/apollo-gateway/src/__tests__/queryPlanCucumber.test.ts

```diff
@@ -0,0 +1,138 @@
+import gql from 'graphql-tag';
+import path from 'path';
+import {
+  GraphQLSchemaModule,
+  GraphQLSchemaValidationError,
+} from 'apollo-graphql';
+import { defineFeature, loadFeature } from 'jest-cucumber';
+import { DocumentNode, GraphQLSchema, GraphQLError } from 'graphql';
+import { composeServices, buildFederatedSchema, normalizeTypeDefs } from '@apollo/federation';
+
+import { QueryPlan } from '../..';
+import { LocalGraphQLDataSource } from '../datasources/LocalGraphQLDataSource';
+import { buildQueryPlan, buildOperationContext } from '../buildQueryPlan';
+
+const buildQueryPlanFeature = loadFeature(
+  './packages/apollo-gateway/src/__tests__/build-query-plan.feature'
+);
+
+
+const features = [
+  buildQueryPlanFeature
+];
+
+function buildLocalService(modules: GraphQLSchemaModule[]) {
+  const schema = buildFederatedSchema(modules);
+  return new LocalGraphQLDataSource(schema);
+}
+
+features.forEach((feature) => {
+  defineFeature(feature, (test) => {
+    feature.scenarios.forEach((scenario) => {
+      test(scenario.title, async ({ given, when, then }) => {
+        let query: DocumentNode;
+        let schema: GraphQLSchema;
+        let errors: GraphQLError[];
+        let queryPlan: QueryPlan;
+        let options: any = {};
+
+        const serviceMap = Object.fromEntries(
+          ['accounts', 'product', 'inventory', 'reviews', 'books', 'documents'].map(
+            serviceName => {
+              return [
+                serviceName,
+                buildLocalService([
+                  require(path.join(
+                    __dirname,
+                    '__fixtures__/schemas',
+                    serviceName,
+                  )),
+                ]),
+              ] as [string, LocalGraphQLDataSource];
+            },
+          ),
+        );
+
+        ({ schema, errors } = composeServices(
+          Object.entries(serviceMap).map(([serviceName, service]) => ({
+            name: serviceName,
+            typeDefs: normalizeTypeDefs(service.sdl()),
+          })),
+        ));
+
+        if (errors && errors.length > 0) {
+          throw new GraphQLSchemaValidationError(errors);
+        }
+
+        const givenQuery = () => {
+          given(/^query$/im, (operation) => {
```

Comment:
Just imagining that this is a basis for porting to other languages, I think providing a type for `operation` here will make it more clear for implementors of other services, perhaps?  i.e.,

```suggestion
          given(/^query$/im, (operation: string) => {
```

(Also, less to fix if we eventually turn on the _no implicit any_ configuration rule for the TypeScript compiler.)